overlord/ifacestate/apparmorprompting/common/common.go

```diff
@@ -0,0 +1,466 @@
+	return time.Now().Format(time.RFC3339Nano)
+}
+
+// NewID returns a new unique ID.
+//
+// The ID is the current unix time in nanoseconds encoded as base32.
+func NewID() string {
+	nowUnix := uint64(time.Now().UnixNano())
+	nowBytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(nowBytes, nowUnix)
+	id := base32.StdEncoding.EncodeToString(nowBytes)
+	return id
+}
+
+// NewIDAndTimestamp returns a new unique ID and corresponding timestamp.
+//
+// The ID is the current unix time in nanoseconds encoded as a string in base32.
+// The timestamp is the same time, encoded as a string in time.RFC3339Nano.
+func NewIDAndTimestamp() (id string, timestamp string) {
+	now := time.Now()
+	nowUnix := uint64(now.UnixNano())
+	nowBytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(nowBytes, nowUnix)
+	id = base32.StdEncoding.EncodeToString(nowBytes)
+	timestamp = now.Format(time.RFC3339Nano)
+	return id, timestamp
+}
+
+// LabelToSnap extracts the snap name from the given AppArmor label.
+//
+// If the label is not of the form 'snap.<snap>.<app>', returns an error, and
+// returns the label as the snap.
+func LabelToSnap(label string) (string, error) {
+	components := strings.Split(label, ".")
+	if len(components) != 3 || components[0] != "snap" {
+		return label, ErrInvalidSnapLabel
+	}
+	snap := components[1]
+	return snap, nil
+}
+
+var (
+	// If kernel request contains multiple interfaces, one must take priority.
+	// Lower value is higher priority, and entries should be in priority order.
+	interfacePriorities = map[string]int{
+		"home":   0,
+		"camera": 1,
+	}
+
+	// List of permissions available for each interface. This also defines the
+	// order in which the permissions should be presented.
+	interfacePermissionsAvailable = map[string][]string{
+		"home":   {"read", "write", "execute"},
+		"camera": {"access"},
+	}
+
+	// A mapping from interfaces which support AppArmor file permissions to
+	// the map between abstract permissions and those file permissions.
+	//
+	// Never include AA_MAY_OPEN in the maps below; it should always come from
+	// the kernel with another permission (e.g. AA_MAY_READ or AA_MAY_WRITE),
+	// and if it does not, it should be interpreted as AA_MAY_READ.
+	interfaceFilePermissionsMaps = map[string]map[string]notify.FilePermission{
+		"home": {
+			"read":    notify.AA_MAY_READ,
+			"write":   notify.AA_MAY_WRITE | notify.AA_MAY_APPEND | notify.AA_MAY_CREATE | notify.AA_MAY_DELETE | notify.AA_MAY_RENAME | notify.AA_MAY_CHMOD | notify.AA_MAY_LOCK | notify.AA_MAY_LINK,
+			"execute": notify.AA_MAY_EXEC | notify.AA_EXEC_MMAP,
+		},
+		"camera": {
+			"access": notify.AA_MAY_WRITE | notify.AA_MAY_READ | notify.AA_MAY_APPEND,
+		},
+	}
+)
+
+// SelectSingleInterface selects the interface with the highest priority from
+// the given list. If none of the given interfaces are included in
+// interfacePriorities, or if the list is empty, return "other".
+func SelectSingleInterface(interfaces []string) string {
```

Comment:
this probably needs to be introduced in the PR that uses it, I'm not sure in isolation whether it's premature or not

Reply:
No, you're right, we don't know anything about how we'll want to go about deciding which interface to use if there are multiple. I'll remove this for now.